sandbox-backend/src/main/java/com/exadel/sandbox/team5/service/impl/DiscountServiceImpl.java

```diff
@@ -18,18 +20,30 @@
 
     private final DiscountDAO discountDAO;
     private final MapperConverter mapper;
+    private final ReviewDAO reviewDAO;
 
     @Override
     public DiscountDto getById(Long id) {
-        return discountDAO.findById(id)
+        DiscountDto discountR = discountDAO.findById(id)
                 .map(discount -> mapper.map(discount, DiscountDto.class))
                 .orElse(null);
+        discountR = setAvarageRate(discountR);
+        return discountR;
+    }
+
+    private DiscountDto setAvarageRate(DiscountDto discountR) {
+        if(discountR == null){
+            return null;
+        }
+        discountR.setRate(reviewDAO.findRate(discountR.getId()));
+        return discountR;
     }
 
     @Override
     public List<DiscountDto> getAll() {
-        List<Discount> discounts = discountDAO.findAll();
-        return mapper.convertList(discounts);
+        List<DiscountDto> resultWithoutRage = mapper.mapAll(discountDAO.findAll(), DiscountDto.class);
+        return
+        resultWithoutRage.stream().map(discount -> discount = setAvarageRate(discount)).collect(Collectors.toList());
```

Comment:
Reformat code (Ctrl + Alt + L in IDE)
Bad lambda usage, ``` .map(discount -> setAvarageRate(discount)) ``` or just ```.map(this::setAvarageRate)``` with method reference

Reply:
changed